src/blocks/services/edit.js

```diff
@@ -1,154 +1,46 @@
-	for ( const style of new TokenList( className ).values() ) {
-		if ( style.indexOf( 'is-style-' ) === -1 ) {
-			continue;
-		}
-
-		const potentialStyleName = style.substring( 9 );
-		const activeStyle = find( styles, { name: potentialStyleName } );
-
-		if ( activeStyle ) {
-			return activeStyle;
-		}
-	}
-
-	return find( styles, 'isDefault' );
-}
-
-/**
- * Replaces the active style in the block's className.
- *
- * @param {string}  className   Class name.
- * @param {Object?} activeStyle The replaced style.
- * @param {Object}  newStyle    The replacing style.
- *
- * @return {string} The updated className.
- */
-function replaceActiveStyle( className, activeStyle, newStyle ) {
-	const list = new TokenList( className );
-
-	if ( activeStyle ) {
-		list.remove( 'is-style-' + activeStyle.name );
-	}
-
-	list.add( 'is-style-' + newStyle.name );
+const Edit = ( props ) => {
+	const {	className, attributes, setAttributes, clientId } = props;
 
-	return list.value;
-}
-
-class Edit extends Component {
-	constructor() {
-		super( ...arguments );
-
-		this.updateStyle = this.updateStyle.bind( this );
-		this.updateInnerAttributes = this.updateInnerAttributes.bind( this );
-		this.onChangeAlignment = this.onChangeAlignment.bind( this );
-		this.onChangeHeadingLevel = this.onChangeHeadingLevel.bind( this );
-		this.toggleCtas = this.toggleCtas.bind( this );
-	}
-
-	componentDidUpdate( prevProps ) {
-		const { attributes, innerBlocks } = this.props;
-		const activeStyle = getActiveStyle( layoutOptions, attributes.className );
-		const lastActiveStyle = getActiveStyle(
-			layoutOptions,
-			prevProps.attributes.className
-		);
-
-		if ( activeStyle !== lastActiveStyle ) {
-			if ( 'circle' === activeStyle.name && ( typeof attributes.alignment === 'undefined' || attributes.alignment === 'none' ) ) {
-				this.onChangeAlignment( 'center' );
-			}
-		}
-
-		if ( innerBlocks.length > 0 ) {
-			const serviceBlocksCount = innerBlocks.reduce( ( acc, cur ) => acc + ( cur.name === 'coblocks/service' ), 0 );
-
-			// Add a new block if the count is less than the columns set.
-			if ( serviceBlocksCount < attributes.columns ) {
-				const { buttons, headingLevel, alignment, insertBlock, clientId } = this.props;
-
-				insertBlock(
-					createBlock( 'coblocks/service', {
-						showCta: buttons,
-						headingLevel,
-						alignment,
-					} ),
-					innerBlocks.length + 1,
-					clientId,
-					false,
-				);
-			}
-		}
-	}
+	const {	updateBlockAttributes, insertBlock, removeBlocks } = useDispatch( 'core/block-editor' );
 
-	updateStyle( style ) {
-		const { className, attributes, setAttributes } = this.props;
+	const { getBlocksByClientId, innerBlocks } = useSelect( ( select ) => {
+		return {
+			getBlocksByClientId: select( 'core/block-editor' ).getBlocksByClientId,
+			innerBlocks: select( 'core/block-editor' ).getBlocks( clientId ),
+			getBlockRootClientId: select( 'core/block-editor' ).getBlockRootClientId,
+		};
+	} );
```

Comment:
Let's "setup" our data within these hooks to be used later rather than returning the actual selectors.

We can remove `getBlockRootClientId` here since it doesn't look like it's being used in this file. The `getBlocksByClientId` can be changed to the `innerItems` or something more descriptive for it's use in the `updateInnerAttributes` and `handlePlaceholderPlacement` helpers below.